### DISCUSSION TITLE
Make button to run script with parameter

### DISCUSSION THREAD

**Discussion by martinszk** [Blackymas/NSPanel_HA_Blueprint#2277] [Ideas]
I have searched the discussion, but I haven't found the answer to my question. I have a script for playing the radio. I have the speaker name set as a variable, and my question is: how can I run the script using a button and pass a parameter to the script?
Thank you for your advice.

**Comment by edwardtfn** [comment]
Are you talking about the hardware button or a virtual button?

**Reply by martinszk** [reply]
About virtual button on Button page. Now I can attach a script to button. It would help me to have one script and use variables to configure it. For example, volume, station, speaker ID, and similar settings.

**Reply by edwardtfn** [reply]
Unfortunately that is not possible right now. You will have to create another script without parameters calling the first one and providing its parameters.
But there are a few enhancement request for buttons which will require us to revisit it some day... What don't you add your as a feature request also?

**Reply by martinszk** [reply]
I wasn't sure if that option was already there and I just couldn't find it. Thanks for your help. Great project!
